ruby/ext/google/protobuf_c/encode_decode.c

```diff
@@ -645,21 +651,19 @@ VALUE Message_decode(VALUE klass, VALUE data) {
 
   const upb_pbdecodermethod* method = msgdef_decodermethod(desc);
   const upb_handlers* h = upb_pbdecodermethod_desthandlers(method);
-  upb_pbdecoder decoder;
-  upb_sink sink;
-  upb_status status = UPB_STATUS_INIT;
+  upb_env e;
+  upb_env_init(&e);
```

Comment:
Cool, let me know if you get the perf back. Also adjust the stack mem size for whatever you are going to create() in the env!